tests/runtime/array

```diff
@@ -166,3 +166,30 @@ PROG struct D { int x; int y[0]; } uprobe:./testprogs/array_access:test_variable
 EXPECT @y: 1
 TIMEOUT 5
 AFTER ./testprogs/array_access
+
+NAME cast pid to int array
+PROG BEGIN { printf("pid byte: %02x\n", ((uint8[8])pid)[0]); exit(); }
+EXPECT pid byte: [0-9a-f][0-9a-f]
```

Comment:
Regexes for expected results are a common pattern in our runtime tests, but in my opinion this kind of test doesn't really add anything.

Checking that we can cast a `pid` to an array and that we can print an array element are both things that unit tests can handle. Checking that the results are correct is what the runtime tests should be doing and a regex doesn't do that.

Reply:
Good point, I dropped the test since there's another one testing correctness of int -> array casting.